api/tests/unit/domain/usecases/find-students-from-certification-center-id_test.js

```diff
@@ -0,0 +1,98 @@
+
+  const certificationCenterId = 1;
+  const wrongCertificationCenterId = 666;
+  const userId = 'userId';
+  let organization;
+
+  const organizationRepository = {
+    getIdByCertificationCenterId: sinon.stub(),
+  };
+  const schoolingRegistrationRepository = {
+    findByOrganizationIdOrderByDivision: sinon.stub(),
+  };
+
+  const certificationCenterMembershipRepository = {
+    doesUserHaveMembershipToCertificationCenter: sinon.stub(),
+  };
+
+  beforeEach(async () => {
+    const externalId = 'AAA111';
+    const certificationCenter = domainBuilder.buildCertificationCenter({ id: certificationCenterId, externalId });
+    organization = domainBuilder.buildOrganization({ externalId });
+
+    organizationRepository.getIdByCertificationCenterId
+      .withArgs(certificationCenter.id).resolves(organization.id);
+
+    certificationCenterMembershipRepository.doesUserHaveMembershipToCertificationCenter
+      .withArgs(userId, certificationCenter.id).resolves(true);
+
+    certificationCenterMembershipRepository.doesUserHaveMembershipToCertificationCenter
+      .withArgs(userId, wrongCertificationCenterId).resolves(false);
+  });
+
+  describe('when user has access to certification center', () => {
+
+    it('should return all students of the linked organization', async () => {
+      // given
+      const expectedStudents = _.times(5, () => domainBuilder.buildSchoolingRegistration({ organization }));
+      schoolingRegistrationRepository.findByOrganizationIdOrderByDivision
+        .withArgs({ organizationId: organization.id }).resolves(expectedStudents);
+
+      // when
+      const studentsFounds = await usecases.findStudentsFromCertificationCenterId({
+        userId,
+        certificationCenterId,
+        organizationRepository,
+        schoolingRegistrationRepository,
+        certificationCenterMembershipRepository,
+      });
+
+      // then
+      expect(studentsFounds).to.be.deep.equal(expectedStudents);
+    });
+
+    context('when the linked organization has no student', () => {
+
+      it('should return empty array', async () => {
+        // given
+        schoolingRegistrationRepository.findByOrganizationIdOrderByDivision
+          .withArgs({ organizationId: organization.id }).resolves([]);
+
+        // when
+        const studentsFounds = await usecases.findStudentsFromCertificationCenterId({
+          userId,
+          certificationCenterId,
+          organizationRepository,
+          schoolingRegistrationRepository,
+          certificationCenterMembershipRepository,
+        });
+
+        // then
+        expect(studentsFounds).to.be.deep.equal([]);
+      });
+    });
+  });
+
+  describe('when user does not have access to certification center', () => {
+    it('should throw an access error', async () => {
+      // when
+      const error = await catchErr(usecases.findStudentsFromCertificationCenterId)({
+        userId,
+        certificationCenterId: wrongCertificationCenterId,
```

Comment:
**Suggestion :** Pourquoi avoir mis cet id dans le context global alors qu'on s'en sert qu'ici ? 
Le mettre dans le `// given` de ce `it` ?

Reply:
Pas de raison particulière 👍